src/lib/components/Form/NumberInput/NumberInput.svelte

```diff
@@ -0,0 +1,142 @@
+{#if label}
+  <div class="number-input-wrapper">
+    {#if controls}
+      <DecrementButton
+        clickHandler={() => value -= step}
+        type={controlsType}
+        disabled={!canDecrement}
+      />
+    {/if}
+    <label>
+      {label}
+      <input {...attr}
+             inputmode="numeric"
+             bind:value={formattedValue}
+             class:border-danger={!isValid && value}
+             class:border-success={isValid && value && valid}
```

Comment:
what is `isValid` and `valid`

Reply:
`valid` is a prop function to add custom validation to the input
`isValid` is a boolean var whose value is computed by calling the `valid` function if provided